include/fastdds/dds/xtypes/dynamic_types/DynamicTypeBuilderFactory.hpp

```diff
@@ -0,0 +1,209 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_BUILDER_FACTORY_HPP
+#define FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_BUILDER_FACTORY_HPP
+
+#include <memory>
+
+#include <fastdds/dds/core/ReturnCode.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/TypeDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/Types.hpp>
+#include <fastrtps/fastrtps_dll.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+namespace xtypes {
+class TypeObject;
+} // namespace xtypes
+
+class RTPS_DllAPI DynamicTypeBuilderFactory : public std::enable_shared_from_this<DynamicTypeBuilderFactory>
+{
+public:
+
+    /*!
+     * Returns the singleton factory object
+     * @remark This method is non thread-safe.
+     * @return @ref DynamicTypeBuilderFactory reference.
```

Comment:
Is it possible for this operation to fail? Also, what is the implementation design in case that the factory has been deleted and then re-instantiated?

Reply:
No, there is no possibility of failing.